backend/prisma/schema.prisma

```diff
@@ -7,28 +7,12 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
-// model Post {
-//   id        Int      @id @default(autoincrement())
-//   createdAt DateTime @default(now())
-//   updatedAt DateTime @updatedAt
-//   title     String   @db.VarChar(255)
-//   content   String?
-//   published Boolean  @default(false)
-//   author    User     @relation(fields: [authorId], references: [id])
-//   authorId  Int
-// }
-
-// model Profile {
-//   id     Int     @id @default(autoincrement())
-//   bio    String?
-//   user   User    @relation(fields: [userId], references: [id])
-//   userId Int     @unique
-// }
+model User {
+  id      Int      @id @default(autoincrement())
```

Comment:
No change necessary but wanted to have a conversation about this for us to think about.

So the type `Int` in figma is the MySQL INT  has a max size of 4,294,967,295. So what would happen if you were to register the 4,294,967,296th user?

It's not really something we need to worry about in this project, (if we got 4 billion users, we should all retire and buy a yacht) but it is something to think about when we make choices like this.

Another consideration to talk about is sequentially incremented ids can lead to some very interesting data vulnerabilities.

For example, lets imagine there is a profile page that has the route of 

`weWrap.com/users/1/profile`

where `1` the id of is Kevin Chin's `User` Row

If I were curious enough to, I might try to directly access

`weWrap.com/users/2/profile` where 2 is Justin's `User` Row

Because they are sequential, I'm about to guess and try to find other peoples profiles, and there is improper security, i might be able to see sensitive data that was not mine.

For reference I think Capitol One had a data leak once because someone found there id number, and then just incremented it by 1 and found other peoples Credit Card numbers.


There are alternative ways of using unique ids (uuids) https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#uuid

This way both overcomes the sequential id numbers as well as increases the maximum possible combinations to nearly infinite.

Reply:
i will change id -> uuid to prevent this security issue